src/view/actions/setOptInPreferences.jsx

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import "regenerator-runtime"; // needed for some of react-spectrum
+import React from "react";
+import Alert from "@react/react-spectrum/Alert";
+import Select from "@react/react-spectrum/Select";
+import RadioGroup from "@react/react-spectrum/RadioGroup";
+import Radio from "@react/react-spectrum/Radio";
+import "@react/react-spectrum/Form"; // needed for spectrum form styles
+import render from "../render";
+import WrappedField from "../components/wrappedField";
+import ExtensionView from "../components/extensionView";
+import getInstanceOptions from "../utils/getInstanceOptions";
+import "./setOptInPreferences.styl";
+
+const purposesEnum = {
+  ALL: "all",
+  NONE: "none"
+};
+
+const getInitialValues = initInfo => {
+  const {
+    instanceName = initInfo.extensionSettings.instances[0].name,
```

Comment:
This is where the magic happens of auto-selecting the first instance. I've made the same kind of change to the other views. Note that we can make the assumption that there will always be at least one instance in the `instances` array because the extension configuration schema enforces at least one instance to always be configured.

Reply:
I love this change!!!